src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilter.java

```diff
@@ -388,11 +416,25 @@ public final void setIdFormat(String format) {
     public boolean accept(AuditEvent event) {
         boolean accepted = true;
         if (event.getLocalizedMessage() != null) {
-            final FileText fileText = getFileText(event.getFileName());
-            if (fileText != null) {
-                final List<Suppression> suppressions = getSuppressions(fileText);
-                accepted = getNearestSuppression(suppressions, event) == null;
+
+            final String fileName = event.getFileName();
+
+            final List<Suppression> suppressions;
+
+            if (event.getSource() instanceof RootModule) {
+                if (!listenerAdded) {
+                    final RootModule checker = (RootModule) event.getSource();
+                    checker.addListener(suppressionsCache);
```

Comment:
@romani Is it ok for a filter to add it's own custom listener?
If it is, shouldn't we official support some type of `init` method instead of having `accept` code do it.

@basinilya It looks like this listener is just to reset the cache when a new file is started.
Why can't you use `event.getFileName()` for this purpose and need the listener instead?

Reply:
> @basinilya It looks like this listener is just to reset the cache when a new file is started.
> Why can't you use `event.getFileName()` for this purpose and need the listener instead?
Because when there's only one file in the project the filename will never change. Yet we need to invalidate cache on next run

Reply:
> Yet we need to invalidate cache on next run

So the only issue is you need to invalidate cache between runs and there is nothing else that specifies if a new run is starting or not?

Reply:
> So the only issue is you need to invalidate cache between runs and there is nothing else that specifies if a new run is starting or not?

Yes.

@rnveach As I wrote in #8807 :
> Or you could add getFileContents() to AuditEvent just like in TreeWalkerAuditEvent

If you don't like adding self a listener I'd prefer the solution above. Then we check if `FileContents` instance is different to invalidate cache. We need to change the signatures of `Checker.fireErrors()` and `Checker.processFile()` to accept `FileContents` instead of file name. `fireErrors()` can be overloaded because the current method it's public.

Reply:
@romani ping